### PR TITLE
Service Accounts - audit for security config change

### PR DESCRIPTION
**Pull request by ywangd**
Add security_config_change auditing for create and delete index-based
service account tokens.

Note this PR is built on top of #72198. So I'd suggest we review and get #72198 through first before looking at this one.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java [hunk 1]
```diff
@@ -1154,6 +1173,34 @@ LogEntryBuilder withRequestBody(DeletePrivilegesRequest deletePrivilegesRequest)
...
         }
 
+        LogEntryBuilder withRequestBody(CreateServiceAccountTokenRequest createServiceAccountTokenRequest) throws IOException {
+            logEntry.with(EVENT_ACTION_FIELD_NAME, "create_service_account_token");
+            XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
+            builder.startObject()
+                .startObject("service_account_token")
+                .field("namespace", createServiceAccountTokenRequest.getNamespace())
+                .field("service", createServiceAccountTokenRequest.getServiceName())
+                .field("token_name", createServiceAccountTokenRequest.getTokenName())
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1183]: Maybe go with only `name`. It's clear from the context whose name it is, and it's the pattern we use elsewhere too.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java [hunk 2]
```diff
@@ -1154,6 +1173,34 @@ LogEntryBuilder withRequestBody(DeletePrivilegesRequest deletePrivilegesRequest)
             return this;
         }
 
+        LogEntryBuilder withRequestBody(CreateServiceAccountTokenRequest createServiceAccountTokenRequest) throws IOException {
+            logEntry.with(EVENT_ACTION_FIELD_NAME, "create_service_account_token");
+            XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
+            builder.startObject()
+                .startObject("service_account_token")
```
Comment (albertzaharovits) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1180]: I noticed that when the token is user it is audited as `service_token` but when it is created or deleted it is `service_account_token`. I don't have any preference but I think it's best to pick one.

File: x-pack/docs/en/security/auditing/event-types.asciidoc [hunk 3]
```diff
@@ -343,15 +355,21 @@ that have been previously described:
...
   `authentication.type`::   Method used to authenticate the user.
                             Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
-  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
+  `apikey.id`          ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                             This attribute is only provided for authentication using an API key.
-  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
+  `apikey.name`        ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                             This attribute is only provided for authentication using an API key.
+  `service_token.name` ::   Name of the <<service-accounts,service account>> token.
+                            This attribute is only provided for authentication using a service account token.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/auditing/event-types.asciidoc:363]: Maybe mention that the service account namespace and service identifier are audited as the `user.name`.

**Review by albertzaharovits — APPROVED** [review]
LGTM, great attention to detail 👍 

I've suggested two field renames, if you accept them take extra care that they're propagated in the `log4j2.properties` file too.

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java (hunk 2 above)
Comment (ywangd) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java:1180]: Thanks for catching it. I meant to use `service_token` for all places that are user visible. But sometimes the word `service_account_token` almost types itself :)

File: x-pack/docs/en/security/auditing/event-types.asciidoc (hunk 3 above)
Comment (ywangd) [reply x-pack/docs/en/security/auditing/event-types.asciidoc:363]: I added a sentence about it under `user.name` where we have a sentence for API key owner as well.

**Review by tvernum — APPROVED** [review]

**Pull request merged**